### PR TITLE
Wait for Transaction Service to announce itself to fix flaky tests

### PR DESCRIPTION
**Pull request by poornachandra**

### CODE REVIEW

File: tephra-core/src/test/java/org/apache/tephra/ThriftTransactionSystemTest.java [hunk 1]
```diff
@@ -88,10 +89,14 @@ protected void configure() {
     txClient = injector.getInstance(TransactionSystemClient.class);
     try {
       LOG.info("Starting transaction service");
+      storage.startAndWait();
```
Comment (anwar6953) [inline tephra-core/src/test/java/org/apache/tephra/ThriftTransactionSystemTest.java]: why start this here, but not in other test cases (TransactionAdminTest, for instance)?

File: tephra-examples/pom.xml [hunk 2]
```diff
@@ -51,6 +55,53 @@
       <artifactId>tephra-core</artifactId>
       <version>${project.version}</version>
     </dependency>
+    <dependency>
```
Comment (anwar6953) [inline tephra-examples/pom.xml]: why were all the pom changes necessary?

Comment (poornachandra) [inline tephra-examples/pom.xml]: Adding HBase dependencies back to tephra-examples pom for IDE

File: tephra-core/src/test/java/org/apache/tephra/ThriftTransactionSystemTest.java (hunk 1 above)
Comment (poornachandra) [inline tephra-core/src/test/java/org/apache/tephra/ThriftTransactionSystemTest.java]: It was being stopped without being started, so for consistency I am starting it here. However, start on this object is a no-op.

**Comment by anwar6953** [comment]
lgtm

**Pull request closed without merging**